### DISCUSSION TITLE
TAO in Debian

### DISCUSSION THREAD

**Discussion by chadsgilbert** [DOCGroup/ACE_TAO#1867] [Q&A]
Hello all, I'm a new user of ACE+TAO starting to work with a well-established app that has used it for some time. Thanks for your work on ACE+TAO.

It looks like TAO was packaged for Debian for a number of years. Of course ACE still is. I am curious why packaging of TAO for Debian has been dropped. Are there technical or other reasons why ACE should be packaged but TAO should not? Or has it just been a lack of contributor availability?

**Comment by jwillemsen** [comment]
As far as I remember it is a licensing issue in TAO_IDL, it has some files that use the Sun Interface Definition Language CFE License which is not accepted by debian packaging. Those licenses are only in the IDL compiler, it doesn't effect the generated code. At some point we had to remove TAO from debian packaging because of that.

**Reply by chadsgilbert** [reply]
Thanks for explaining, @jwillemsen. I came across a thread today with yourself and Ken Sedgwick from ca. 2008 with Fedora that seems to confirm exactly what you say here.

The license issue looks to come from an old Sun Microsystems (now Oracle IP?) license on the IDL compiler frontend ("CFE") imposing some constraints that allow distribution, but which (in Fedora legal's opinion at the time) do not qualify as Free Software. That sounds like something that would prevent distribution as part of core, but couldn't it be distributed in non-free?

Perhaps that's a better question for Debian, but I'll ask you here first in case you remember the answer.

**Reply by jwillemsen** [reply]
Yes, some old code which has a Sun license. We tried to get that code relicensed but never got a reply of the people we contacted. 

Not sure about where TAO could be packaged, when there is a non-free part, that sounds a good place for TAO, the generated code itself (which is linked into your application) is again free.

**Reply by chadsgilbert** [reply]
This actually raises another question - would it be desirable (and in accordance with licenses) to release TAO run-time libraries even if the development libraries are not releasable in that way?

**Reply by jwillemsen** [reply]
I am personally not in favor of pre build runtime libraries as there are so many options to tune TAO. Very likely different users need different settings.

Some years ago as part of a project we created a IDL to C++ backend for [RIDL](https://github.com/RemedyIT/ridl) which could replace TAO_IDL but that work was never released back into open source, that would have solved this licensing issue but also made cross compilation much easier.

**Reply by chadsgilbert** [reply]
Okay, it sounds like packaging TAO is not the right thing to do, and distributing the resulting runtime as part of an application is more suitable, given the licensing constraints and complexity/configurability of the runtime libraries.

Thanks for talking this through with me. We didn't come to the answer I hoped for, but it's much clearer why things are the way they are.

**Reply by ophiuchus1980** [reply]
hello. please Guide me   debian12     How to configure  ace+tao   library      In which file is it configured     What line is the configuration in this file. Could you please provide more details      My English is very poor, I am learning slowly, sorry

**Comment by ophiuchus1980** [comment]
hello. please Guide me   debian12     How to configure  ace+tao   library      In which file is it configured     What line is the configuration in this file. Could you please provide more details      My English is very poor, I am learning slowly, sorry
